mobile/src/reducers/calendar/specs/intervals-by-single-day-selection.epic.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { ActionsObservable } from 'redux-observable';
+import moment from 'moment';
+import { Observable } from 'rxjs';
+import { DayModel } from '../calendar.model';
+import { CalendarEvent, DatesInterval, CalendarEventStatus, CalendarEventType } from '../calendar-event.model';
+import { CalendarEvents } from '../calendar-events.model';
+import { intervalsBySingleDaySelectionEpic$ } from '../calendar.epics';
+import { SelectCalendarDay, LoadCalendarEventsFinished, CalendarEventCreated, selectCalendarDay, loadCalendarEventsFinished, calendarEventCreated, selectIntervalsBySingleDaySelection } from '../calendar.action';
+
+describe('intervalsBySingleDaySelectionEpic', () => {
+    let action$: ActionsObservable<SelectCalendarDay | LoadCalendarEventsFinished | CalendarEventCreated>;
+    beforeEach(() => {
+        const day: DayModel = {
+            date: moment(),
+            today: true,
+            belongsToCurrentMonth: true
+        };
+
+        const loadedCalendarEvent = new CalendarEvent();
+
+        loadedCalendarEvent.calendarEventId = '1';
+        loadedCalendarEvent.dates = new DatesInterval();
+        loadedCalendarEvent.dates.startDate = moment();
+        loadedCalendarEvent.dates.endDate = moment(loadedCalendarEvent.dates.startDate);
+        loadedCalendarEvent.status = CalendarEventStatus.Requested;
+        loadedCalendarEvent.type = CalendarEventType.Sickleave;
+
+        const calendarEvents = new CalendarEvents([loadedCalendarEvent]);
+
+        const createdCalendarEvent = new CalendarEvent();
+
+        createdCalendarEvent.calendarEventId = '1';
+        createdCalendarEvent.dates = new DatesInterval();
+        createdCalendarEvent.dates.startDate = moment();
+        createdCalendarEvent.dates.endDate = moment(createdCalendarEvent.dates.startDate);
+        createdCalendarEvent.status = CalendarEventStatus.Requested;
+        createdCalendarEvent.type = CalendarEventType.Sickleave;
+
+        action$ = new ActionsObservable(Observable.concat(
+            Observable.of(selectCalendarDay(day)),
+            Observable.of(loadCalendarEventsFinished(new CalendarEvents([loadedCalendarEvent]))),
+            Observable.of(calendarEventCreated(createdCalendarEvent))
+        ));
+    });
+
+    it('should select intervals by single day selection', (done) => {
+        intervalsBySingleDaySelectionEpic$(action$).subscribe(x => {
```

Comment:
does it unsubscribe automatically?

Reply:
when tests are unloaded

Reply:
what tests? all tests?

Reply:
`Observable.of` is completed observable, listeners automatically will be removed after subscribe was fired.
```
  protected _subscribe(subscriber: Subscriber<T>): TeardownLogic {
    const value = this.value;
    const scheduler = this.scheduler;

    if (scheduler) {
      return scheduler.schedule(ScalarObservable.dispatch, 0, {
        done: false, value, subscriber
      });
    } else {
      subscriber.next(value);
      if (!subscriber.closed) {
        subscriber.complete();
      }
    }
  }
...
  complete(): void {
    if (!this.isStopped) {
      this.isStopped = true;
      this._complete();
    }
  }
  protected _complete(): void {
    this.destination.complete();
    this.unsubscribe();
  }
```
if you even invoke unsubscribe on completed observable, it will do nothing:
```
  unsubscribe(): void {
    if (this.closed) {
      return; // <------
    }
    this.isStopped = true;
    super.unsubscribe();
  }
```